pkg/trustedresources/verify.go

```diff
@@ -39,8 +39,27 @@ import (
 const (
 	// SignatureAnnotation is the key of signature in annotation map
 	SignatureAnnotation = "tekton.dev/signature"
+	IgnoreNoMatchPolicy = iota
+	WarnNoMatchPolicy
+	VerificationPass
+	WarnModeVerificationFail
 )
 
+// VerificationResultType indicates different cases of a verification result when nil error is returned for verification.
+type VerificationResultType int
+
+// VerificationResult contains the type and message of cases when the verification doesn't return error.
+type VerificationResult struct {
+	// VerificationResultType has 4 types which is corresponding to 4 cases:
+	// 1) IgnoreNoMatchPolicy: The verification is skipped when there's no matching policies.
+	// 2) WarnNoMatchPolicy: There's no matching policies and warning is logged.
+	// 3) VerificationPass: The verification passes.
+	// 4) WarnModeVerificationFail: Only Warn mode verification policies fail.
+	VerificationResultType VerificationResultType
+	// Message contains the warning message when there is a warning logged.
```

Comment:
just curious, is this a k8s style "Reason" e.g. "NoMatchingVerificationPolicies" or meant to be human-readable?

Reply:
I think this meant to be human-readable, my plan is to update the condition message with this message